tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/GraphQL/HotChocolate/HotChocolateCommon.cs

```diff
@@ -160,6 +166,121 @@ private static string ConstructErrorMessage(List<IError> executionErrors)
             return Util.StringBuilderCache.GetStringAndRelease(builder);
         }
 
+        private static List<SpanEvent> ConstructErrorEvents(List<IError> executionErrors)
+        {
+            List<SpanEvent> spanEvents = [];
+
+            try
+            {
+                for (int i = 0; i < executionErrors.Count; i++)
+                {
+                    var eventAttributes = new List<KeyValuePair<string, object>>();
+                    var executionError = executionErrors[i];
+
+                    var message = executionError.Message;
+                    if (message != null)
+                    {
+                        eventAttributes.Add(new KeyValuePair<string, object>("message", message));
+                    }
+
+                    var locations = executionError.Locations;
+                    if (locations != null)
+                    {
+                        var joinedLocations = new List<string>();
+                        foreach (var location in locations)
+                        {
+                            if (location.TryDuckCast<ErrorLocationStruct>(out var locationProxy))
+                            {
+                                joinedLocations.Add($"{locationProxy.Line}:{locationProxy.Column}");
+                            }
+                        }
+
+                        eventAttributes.Add(new KeyValuePair<string, object>("locations", joinedLocations.ToArray()));
+                    }
+
+                    var path = executionError.Path;
+                    if (path.Name != null)
+                    {
+                        eventAttributes.Add(new KeyValuePair<string, object>("path", new[] { path.Name }));
+                    }
+
+                    var code = executionError.Code;
+                    if (code != null)
+                    {
+                        eventAttributes.Add(new KeyValuePair<string, object>("code", code));
+                    }
+
+                    var exception = executionError.Exception;
+                    if (exception != null)
+                    {
+                        eventAttributes.Add(new KeyValuePair<string, object>("stacktrace", exception.StackTrace));
+                    }
+
+                    var extensions = executionError.Extensions;
+                    if (extensions is { Count: > 0 })
+                    {
+                        var configuredExtensions = Tracer.Instance.Settings.GraphQLErrorExtensions;
+
+                        foreach (var extension in extensions)
+                        {
+                            if (configuredExtensions.Contains(extension.Key))
+                            {
+                                var key = extension.Key;
+                                var value = extension.Value;
```

Comment:
I think it would be better to use a null-coalescing here:
```suggestion
                                var value = extension.Value ?? "null";
```
So we remove save lines below 🙂